third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/TMEMAlloc1D.cpp

```diff
@@ -165,18 +148,34 @@ class TMEM1DAllocator {
   }
 
 public:
-  void replaceWith1DTMEM(
-      Operation *producer, Operation *consumer,
-      std::optional<ttng::TMEMAllocOp> allocOpBuffer = std::nullopt) {
+  void
+  replaceWith1DTMEM(Operation *producer, Operation *consumer,
+                    std::optional<Operation *> allocOpBuffer = std::nullopt) {
     this->numWarps = ttg::lookupNumWarps(producer);
     assert((numWarps == 4 || numWarps == 8) && "Only support 4 or 8 warps");
     TMEMStore1D(producer, allocOpBuffer);
     TMEMLoad1D(producer, consumer);
   }
 };
 
-void generate1DAllocations(OpBuilder &builder, Operation *producer) {
+void generate1DAllocations(OpBuilder &builder, Operation *producer,
+                           llvm::SmallVector<ttng::TMEMAllocOp> &allocOps) {
   assert(producer->hasAttr("tmem.start") && "Expected tmem.start");
+  std::optional<Operation *> allocOpBuffer = std::nullopt;
+  auto producerTMEMStart =
+      mlir::cast<mlir::IntegerAttr>(producer->getAttr("tmem.start")).getInt();
+  if (producerTMEMStart < allocOps.size()) {
+    auto allocOp = allocOps[producerTMEMStart];
+    auto allocShape = allocOp.getType().getShape();
+    if (allocShape[1] != 1) {
+      builder.setInsertionPointAfter(allocOp);
+      // Hardcode allocShape[0] / 2 for testing.
```

Comment:
This is to match the IR where its specified in gluon. Not sure how we are going to determine the offset.